docs/examples/grpc/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM golang as build
+
+ADD https://raw.githubusercontent.com/grpc/grpc-go/master/examples/features/reflection/server/main.go /
+
+RUN apt update && \
+  apt upgrade -y && \
+  apt install -y unzip && \
+  mkdir -p /home/downloads/protoc && \
+  cd /home/downloads/protoc  && \
+  PB_REL="https://github.com/protocolbuffers/protobuf/releases" && \
+  curl -LO $PB_REL/download/v3.15.8/protoc-3.15.8-linux-x86_64.zip && \
+  unzip protoc-3.15.8-linux-x86_64.zip && \
```

Comment:
We can just using ADD command to do this.

Reply:
ok .. fixing these and more now

Reply:
@tao12345666333 , i just realized that ADD will add a new layer to the image. Please advise if adding a new layer to the image is preferred choice over handling it all in the RUN layer of the image.

Reply:
Because this is using multi-stage build, the final image is not affected by this.

Reply:
ah, right. understood. Slow witted me.

Reply:
I think we can "simplify" this a bit :) We don't need protoc, etc.

What we can do here is:
* curl https://github.com/grpc/grpc-go/blob/v1.39.x/examples/helloworld/greeter_server/main.go
* compile only this

Because this program uses an already compiled protobuf spec (https://github.com/grpc/grpc-go/blob/v1.39.x/examples/helloworld/greeter_server/main.go#L28) we don't need to compile it again.

And the good news is that golang builder image already have curl.

So something like:
```
WORKDIR /go/src
RUN curl https://github.com/grpc/grpc-go/blob/v1.39.x/examples/helloworld/greeter_server/main.go && \
        go build -o /greeter-server main.go
```

should be enough

Reply:
Chaned. will appear in next squashed push